### PR TITLE
chore(devex): use hatch lint:style instead of individual scripts in commit hooks

### PR DESCRIPTION
**Pull request by romainkomorndatadog**
Updates Git commit hooks to match CI pre-check use hatch-provide lint:style check.

Currently, `black` and `isort` will reformat files during the `pre-commit` hooks run, which will result in files being committed that will fail the CI `pre_check` (because the changed files need to be committed prior to push).

This also introduces the `lint:fmt` script that runs `isort` and `black`, as well as `run-fmt.sh` convenience util script (though it's not added as a pre-commit hook).

### Introduce changes
```
romain.komorn@COMP-CLXFH7L2J3 dd-trace-py % git diff
diff --git a/ddtrace/contrib/pytest/plugin.py b/ddtrace/contrib/pytest/plugin.py
index 0daa97fb3..d3169cfbb 100644
--- a/ddtrace/contrib/pytest/plugin.py
+++ b/ddtrace/contrib/pytest/plugin.py
@@ -27,10 +27,10 @@ from ddtrace.contrib.pytest.constants import HELP_MSG
 from ddtrace.contrib.pytest.constants import KIND
 from ddtrace.contrib.pytest.constants import XFAIL_REASON
 from ddtrace.contrib.unittest import unpatch as unpatch_unittest
+from ddtrace.internal.ci_visibility.constants import COVERAGE_TAG_NAME
 from ddtrace.ext import SpanTypes
 from ddtrace.ext import test
 from ddtrace.internal.ci_visibility import CIVisibility as _CIVisibility
-from ddtrace.internal.ci_visibility.constants import COVERAGE_TAG_NAME
 from ddtrace.internal.ci_visibility.constants import EVENT_TYPE as _EVENT_TYPE
 from ddtrace.internal.ci_visibility.constants import MODULE_ID as _MODULE_ID
 from ddtrace.internal.ci_visibility.constants import MODULE_TYPE as _MODULE_TYPE
@@ -58,12 +58,14 @@ def encode_test_parameter(parameter):

     param_repr = repr(parameter)
     # if the representation includes an id() we'll remove it
     # because it isn't constant across executions
-    return re.sub(r" at 0[xX][0-9a-fA-F]+", "", param_repr)
+    return re.sub(r" at 0[xX][0-9a-fA-F]+",
+                  "", param_repr)


 def is_enabled(config):
     """Check if the ddtrace plugin is enabled."""
-    return (config.getoption("ddtrace") or config.getini("ddtrace")) and not config.getoption("no-ddtrace")
+    return (config.getoption("ddtrace")
+            or config.getini("ddtrace")) and not config.getoption("no-ddtrace")
```

### Fail to commit
```
romain.komorn@COMP-CLXFH7L2J3 dd-trace-py % git commit -am "testing"
[Autohook] Looking for pre-commit scripts to run...found 4!
[Autohook] BEGIN 01-style
───────────────────────────────────────────────────────────────────────────────────────── lint ──────────────────────────────────────────────────────────────────────────────────────────
Finished checking dependencies
cmd [1] | isort --check --diff ddtrace/contrib/pytest/plugin.py
ERROR: /Users/romain.komorn/go/src/github.com/DataDog/dd-trace-py/ddtrace/contrib/pytest/plugin.py Imports are incorrectly sorted and/or formatted.
--- /Users/romain.komorn/go/src/github.com/DataDog/dd-trace-py/ddtrace/contrib/pytest/plugin.py:before	2023-09-04 13:35:14.005828
+++ /Users/romain.komorn/go/src/github.com/DataDog/dd-trace-py/ddtrace/contrib/pytest/plugin.py:after	2023-09-04 13:35:40.203034
@@ -27,10 +27,10 @@
 from ddtrace.contrib.pytest.constants import KIND
 from ddtrace.contrib.pytest.constants import XFAIL_REASON
 from ddtrace.contrib.unittest import unpatch as unpatch_unittest
-from ddtrace.internal.ci_visibility.constants import COVERAGE_TAG_NAME
 from ddtrace.ext import SpanTypes
 from ddtrace.ext import test
 from ddtrace.internal.ci_visibility import CIVisibility as _CIVisibility
+from ddtrace.internal.ci_visibility.constants import COVERAGE_TAG_NAME
 from ddtrace.internal.ci_visibility.constants import EVENT_TYPE as _EVENT_TYPE
 from ddtrace.internal.ci_visibility.constants import MODULE_ID as _MODULE_ID
 from ddtrace.internal.ci_visibility.constants import MODULE_TYPE as _MODULE_TYPE
[Autohook] FINISH 01-style
[Autohook] BEGIN 02-run-mypy
───────────────────────────────────────────────────────────────────────────────────────── lint ──────────────────────────────────────────────────────────────────────────────────────────
Finished checking dependencies
cmd [1] | mypy
ddtrace/appsec/_deduplications.py:14: note: By default the bodies of untyped functions are not checked, consider using --check-untyped-defs  [annotation-unchecked]
ddtrace/debugging/_probe/remoteconfig.py:316: note: By default the bodies of untyped functions are not checked, consider using --check-untyped-defs  [annotation-unchecked]
Success: no issues found in 545 source files
[Autohook] FINISH 02-run-mypy
[Autohook] BEGIN 03-run-codespell
───────────────────────────────────────────────────────────────────────────────────────── lint ──────────────────────────────────────────────────────────────────────────────────────────
Finished checking dependencies
cmd [1] | codespell ddtrace/contrib/pytest/plugin.py
[Autohook] FINISH 03-run-codespell
[Autohook] BEGIN 04-run-clang-format
Run clang-format skipped: No C++ files were found in `git diff --staged`
[Autohook] FINISH 04-run-clang-format
[Autohook] A pre-commit script yielded negative exit code 1
```

### Use new `fmt` script:
```
romain.komorn@COMP-CLXFH7L2J3 dd-trace-py % hatch -v run lint:fmt
───────────────────────────────────────────────────────────────────────────────────────── lint ──────────────────────────────────────────────────────────────────────────────────────────
Finished checking dependencies
cmd [1] | black .
reformatted ddtrace/contrib/pytest/plugin.py
All done! ✨ 🍰 ✨
1 file reformatted, 1284 files left unchanged.
cmd [2] | isort .
Fixing /Users/romain.komorn/go/src/github.com/DataDog/dd-trace-py/ddtrace/contrib/pytest/plugin.py
Skipped 25 files
```
### See that we're back to pre-testing state:
```
romain.komorn@COMP-CLXFH7L2J3 dd-trace-py % git diff
romain.komorn@COMP-CLXFH7L2J3 dd-trace-py %
```

## Checklist

- [x] Change(s) are motivated and described in the PR description.
- [x] Testing strategy is described if automated tests are not included in the PR.
- [x] Risk is outlined (performance impact, potential for breakage, maintainability, etc).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] [Library release note guidelines](https://ddtrace.readthedocs.io/en/stable/releasenotes.html) are followed. If no release note is required, add label `changelog/no-changelog`.
- [x] Documentation is included (in-code, generated user docs, [public corp docs](https://github.com/DataDog/documentation/)).
- [x] Backport labels are set (if [applicable](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting))

## Reviewer Checklist

- [x] Title is accurate.
- [x] No unnecessary changes are introduced.
- [x] Description motivates each change.
- [x] Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes unless absolutely necessary.
- [x] Testing strategy adequately addresses listed risk(s).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] Release note makes sense to a user of the library.
- [x] Reviewer has explicitly acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment.
- [x] Backport labels are set in a manner that is consistent with the [release branch maintenance policy](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting)
- [x] If this PR touches code that signs or publishes builds or packages, or handles credentials of any kind, I've requested a review from `@DataDog/security-design-and-guidance`.
- [x] This PR doesn't touch any of that.

### CODE REVIEW

**Comment by romainkomorndatadog** [comment]
I think this is more high-friction than trying to automatically format and commit, or automatically format and fail (which as far as I can tell, would require running `black` twice), but it's also more explicit, which I prefer.

File: .github/workflows/apm-transport-stress-test.yml [hunk 1]
```diff
@@ -13,7 +13,7 @@ jobs:
       DD_API_KEY: ${{ secrets.DD_SHARED_TESTS_API_KEY }}
       TRACER: python
     steps:
-      - uses: actions/checkout@v3
+      - uses: actions/checkout@v4
```
Comment (brettlangdon) [inline .github/workflows/apm-transport-stress-test.yml:16]: can you move these bumps into their own PR?

**Review by brettlangdon — COMMENTED** [review]
I'd prefer the github action bump to be it's own PR, otherwise lgtm

File: .github/workflows/apm-transport-stress-test.yml (hunk 1 above)
Comment (romainkomorndatadog) [reply .github/workflows/apm-transport-stress-test.yml]: Done. Also, I'm not sure it really needs to happen anyway.

**Review by brettlangdon — APPROVED** [review]

**Review by majorgreys — APPROVED** [review]
